Mage/src/main/java/mage/abilities/effects/ContinuousEffect.java

```diff
@@ -54,6 +59,20 @@ public interface ContinuousEffect extends Effect {
 
     List<MageObjectReference> getAffectedObjects();
 
+    List<MageObject> queryAffectedObjects(Game game, Ability source);
+
+    void clearDynamicAffectedObjects();
+
+    EnumSet<Zone> getEffectCardZones();
+
+    FilterStackObject getAffectedStackObjectFilter();
+
+    FilterCard getAffectedCardFilter();
```

Comment:
It doesn’t help to find really affected objects - xmage effects uses “apply” and custom code logic inside it, not filters only. There 600 effects to modify to support it.